src/RequestConditionalPlugin.php

```diff
@@ -0,0 +1,46 @@
+<?php
+
+namespace Http\Client\Plugin;
+
+use Http\Message\RequestMatcher;
+use Psr\Http\Message\RequestInterface;
+
+/**
+ * Apply a delegated plugin based on a request match.
+ *
+ * @author Márk Sági-Kazár <mark.sagikazar@gmail.com>
+ */
+final class RequestConditionalPlugin implements Plugin
+{
+    /**
+     * @var RequestMatcher
+     */
+    private $requestMatcher;
+
+    /**
+     * @var Plugin
+     */
+    private $delegatedPlugin;
+
+    /**
+     * @param RequestMatcher $requestMatcher
+     * @param Plugin         $delegatedPlugin
+     */
+    public function __construct(RequestMatcher $requestMatcher, Plugin $delegatedPlugin)
```

Comment:
I was also thinking about that. For that case I would add a separate
ChainPlugin.

Tobias Nyholm <notifications@github.com> ezt írta (2016. március 5.,
szombat):

> In src/RequestConditionalPlugin.php
> <https://github.com/php-http/plugins/pull/65#discussion_r55119904>:
>
> > +{
> > +    /**
> > +     * @var RequestMatcher
> > +     */
> > +    private $requestMatcher;
> > +
> > +    /**
> > +     * @var Plugin
> > +     */
> > +    private $delegatedPlugin;
> > +
> > +    /**
> > +     * @param RequestMatcher $requestMatcher
> > +     * @param Plugin         $delegatedPlugin
> > +     */
> > +    public function __construct(RequestMatcher $requestMatcher, Plugin $delegatedPlugin)
>
> I maybe want to use more than one plugin. I suggest allowing the second
> parameter to be an array of Plugins. Would that be possible?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/php-http/plugins/pull/65/files#r55119904>.
>